### ISSUE TITLE
Enable authentication for web sites

### ISSUE THREAD

**Issue by NLopezPerez** [danswer-ai/danswer#502]

**Comment by Weves** [comment]
Hey @NLopezPerez, can you add a little more context as to what you're looking for here?

**Comment by L-Chris** [comment]
auto add cookie to web sites

**Comment by evin-bz** [comment]
I think this is a dupe of: https://github.com/danswer-ai/danswer/issues/796

Without much more info I'd close this

**Comment by zarlor** [comment]
If I had to guess (and potentially add this as a vote for this as a feature request as well) I think this is not a duplicate. That one is about login into Danswer, but I believe this request would be for, for example, going to a web page with the web connector that will only display the page if a user is logged into that site. If I wanted to get a page (or set of pages) scraped from a forum that will only show that after login then a way to send the login (or the cookie with the cached credentials) would facilitate that situation.

**Comment by anthonyscata** [comment]
I have the same use case. I have a number of web pages that are restricted behind basic auth or a token. Would be great to supply credentials which are used by the web connector to authenticate and scrape / index pages as for now I am restricted to only public pages or indirect workarounds.

**Comment by AlvaroRojas** [comment]
Adding credential support for web connector would be huge (cookie, auth token header...)

**Comment by nigelsim** [comment]
Agreed. This would be a nice simple way to integrate some of our custom back office apps. Using a sitemap as the index.

Looking at [the code](https://github.com/onyx-dot-app/onyx/blob/7af10308d7c658d7162e41fcd27899d746becd83/backend/onyx/connectors/web/connector.py#L313) it looks like there is some kind of OAuth support already. Just not configurable from the UI I guess.

Also related to #859
